newDB.sql

```diff
@@ -1,97 +1,110 @@
+  status varchar(255) not null,
+  type text not null,
+  balance float not null,
+  created_at timestamp without time zone not null default now(),
+  updated_at timestamp without time zone,
+  check(created_at < updated_at)
 );
 
-create table Employee (
-  Emp_ID int not null primary key,
-  Emp_Name varchar(255) not null,
-  Login_ID varchar(255) not null,
-  password varchar(25) not null,
-  Email_ID varchar(255) not null,
-  Contact_No varchar(25) not null,
-  Create_Date date not null,
-  Last_Login date not null
-);
+create index accounts_customer_id_index on Accounts(customer_id);
 
-create table Loan (
-  Loan_ID int not null primary key,
-  Loan_Type varchar(25) not null,
-  Loan_Amount varchar(25) not null,
-  Customer_ID int not null references Customer(CustomerID),
-  Interest float not null,
-  Start_Date date not null
+create table Loans (
+  id bigserial primary key,
+  customer_id bigint not null references Customers(id),
+  type text not null,
+  amount float not null,
+  interest float not null,
+  period float not null,
+  balance float not null,
+  created_at timestamp without time zone null default now(),
+  monthly_emi float not null,
+  monthly_emi_remaining int not null,
+  monthly_emi_date date not null,
+  updated_at timestamp without time zone,
+  check (balance <= amount),
+  check (updated_at > created_at)
+  check (monthly_emi_remaining < period)
 );
 
-create table Loan_Payment (
-  Payment_ID int not null primary key,
-  Date date not null,
-  Paid_Amount float not null,
-  Principle_Amount float not null,
-  Balance float not null
+create index loans_customer_id_index on Loans(customer_id);
+
+--check for prepaid??
+create table Loan_Payments (
+  id bigserial primary key,
+  loan_id bigint not null references Loans(id),
+  payment_id varchar(255) not null unique,
+  date timestamp without time zone not null,
+  amount float not null,
+  prepaid boolean not null
 );
 
-create table Loan_type (
-  Loan_Type varchar(25) not null,
-  prefix varchar(25) not null,
-  Max_Amount float not null,
-  Min_Amount float not null,
-  Interest float not null,
-  status varchar(25) not null
+create index loan_payments_loan_id_index on Loan_Payments(loan_id);
+
+create table Transactions (
+  id bigserial primary key,
+  sender_id bigint not null references Accounts(id),
+  receiver_id bigint not null references Accounts(id),
+  amount float not null,
+  status text not null,
+  created_at timestamp without time zone default now() not null
 );
 
-create table Register_Payee (
-  SL_No decimal not null primary key,
-  Payee_Name varchar(25) not null,
-  Account_No varchar(25) not null references Accounts(Account_No),
-  Account_Type varchar(25) not null,
-  Bank_Name varchar(25) not null,
-  IFS_Code varchar(25) not null  references Branch(IFS_Code)
+create index transactions_sender_id_index on Transactions(sender_id);
+create index transactions_receiver_id_index on Transactions(receiver_id);
+
+create table Withdraws (
```

Comment:
This columns purpose is overlapping with transactions table, give it another thought

Reply:
Sir transactions table is for the transfer between 2 different accounts and withdraws and deposits are the transactions made in the self-account if anyone wants to query only the withdraws or deposits then they can access it using these tables.